examples/timebase_master.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#  timebase_master.py
+#
+"""A simple JACK timebase master."""
+
+import argparse
+import sys
+from threading import Event
+
+import jack
+
+
+class TimebaseMasterClient(jack.Client):
+    def __init__(self, name, bpm=120.0, meter='4/4', beats_per_bar=4,
+                 beat_type=4, ticks_per_beat=1920, conditional=False,
+                 debug=False, **kw):
+        super().__init__(name, **kw)
+        self.beats_per_bar = beats_per_bar
+        self.beat_type = beat_type
+        self.bpm = bpm
+        self.conditional = conditional
+        self.debug = debug
+        self.ticks_per_beat = int(ticks_per_beat)
```

Comment:
That doesn't make much sense to me, musically, but I can convert the instance attribute via `float()` when setting the `pos` attribute.

Also, since tick is an `int32_t`, we need to be careful when assigning expressions to it that involve `ticks_per_beat`, that the type of the expression result is an `int`, otherwise `cffi` will throw an error.

I noticed that `beats_per_bar` and `beat_type` are also of type `float` (again, makes no sense to me, musically), so I changed the code to convert these as well when assigning to `pos`, while keeping the instance attributes as type `int`: e5c018e40d5ad75328f5dc723aefd02a7bbfdac5

Reply:
> That doesn't make much sense to me, musically

Probably not, probably it does?

I would simply allow `float` values because JACK allows them.

> we need to be careful when assigning expressions to it

Yes, but it doesn't help if we assign an `int` to `ticks_per_beat`, because it will be immediately converted to a `double`.

> again, makes no sense to me, musically

Again, I don't know, I would still allow it if JACK thinks that's a good idea.

> keeping the instance attributes as type `int`

I would just pass it through to the JACK functions, CFFI will check it and complain if it isn't right.

I think converting the inputs with `int()` is not a good idea, because it silently converts user-provided `float` values.
If you really want to require `int`s (which I think you shouldn't), you should at least fail loudly when the uses tries to use a `float` value.

Also, it's very confusing if your local copy of the values have different data types than the values in the `pos` struct, isn't it?